server/controllers/submission.js

```diff
@@ -0,0 +1,146 @@
+const Submission = require("../models/Submission");
+const Contest = require("../models/Contest");
+const asyncHandler = require("express-async-handler");
+const uploadMultiImage = require("../utils/uploadFiles");
+
+exports.createSubmission = asyncHandler(async (req, res, next) => {
+    
+    const { creator, contest_id, is_active, date_Created } = req.body;
+   
+    let contest = await Contest.findById(contest_id);
+
+    //Grabbing AWS URL to import into submission model
+    multiImageLocationArray = [];
+    let fileArray = req.files,
+        fileLocation;
+    for (let i = 0; i < fileArray.length; i++) {
+        fileLocation = fileArray[i].location;
+        multiImageLocationArray.push(fileLocation);
+    }
+    let files = multiImageLocationArray
+    //Creating a new submission with aws url array and submission data received from front end
+    if (contest) {
+        const createdSubmission = await Submission.create({
+            creator,
+            contest_id,
+            files,
+            is_active,
+            date_Created,
+        });
+        //Creating a finding and updating contest to apply submission id to array within contest
+        if (createdSubmission) {
+            let submissionArr = contest.submissions;
+            submissionArr.push(createdSubmission._id);
+            contest.submissions = submissionArr;
+
+            const title =contest.title;
+            const price =contest.price;
+            const description = contest.description;
+            const end_date = contest.end_date
+            const images = contest.images;
+            const submissions = contest.submissions;
+            //
+            let createdContest = await Contest.findByIdAndUpdate(
+                contest_id,
+                {
+                    title,
+                    price,
+                    description,
+                    end_date,
+                    images,
+                    submissions,
+                },
+                { new: true }
+            );
+            if (createdContest) {
+                console.log("successfully updated contest");
+            } else {
+                console.log("failed to update contest");
+            }
+
+            res.status(201).json({
+                success: {
+                    submission: createdSubmission,
+                },
+            });
+        } else {
+            res.status(500);
+            throw new Error(
+                "Submission could not be processed at this time, Please Try Again"
+            );
+        }
+    } else {
+        res.status(500);
+        throw new Error(
+            "Contest not found to allow submission"
```

Comment:
should the error message be that the Contest does not exist?

Reply:
I agree the error message should be contest does not exist